src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -65,7 +65,7 @@ public Http2Stream(HttpRequestMessage request, Http2Connection connection, int s
                 _state = StreamState.ExpectingHeaders;
 
                 _request = request;
-                _response = new HttpResponseMessage()
+                _response = new HttpResponseMessage(0)
```

Comment:
Why change this?  Why pass in '0' which is really an HttpStatusCode?

Reply:
are you suggesting 0 is valid code normal server would send?

Reply:
No, 0 is not a valid status code usually.  I don't understand why you're changing this line in the first place.

Reply:
I wanted to track if valid status code was received. So I set this to 0 (e.g. invalid) and than when  status code is received it will get proper value.
And I can use (StatusCode == 0) in other places to know if headers were received out of order.

Reply:
You should use named arguments here to help readability, i.e. 

```c#
_response = new HttpResponseMessage(statusCode: 0);
```

Also, it would be helpful in future PRs if you could add explanation of your design changes (like this tracking against statusCode=0 for errors) into your commit/PR description.  It will help speed up review by others.